src/core/thread/network_data_leader.cpp

```diff
@@ -132,6 +132,29 @@ Error Leader::GetPreferredNat64Prefix(ExternalRouteConfig &aConfig) const
     return error;
 }
 
+bool Leader::IsNat64(const Ip6::Address &aAddress) const
+{
+    bool                isNat64  = false;
+    Iterator            iterator = kIteratorInit;
+    ExternalRouteConfig config;
+
+    while (GetNextExternalRoute(iterator, config) == kErrorNone)
+    {
+        if (!config.mNat64 || !config.GetPrefix().IsValidNat64())
+        {
+            continue;
+        }
+
+        if (aAddress.MatchesPrefix(config.GetPrefix()))
```

Comment:
```suggestion
        if (config.mNat64 && config.GetPrefix().IsValidNat64() && aAddress.MatchesPrefix(config.GetPrefix()))
```
What about writing like this?

Reply:
Changed in new push. Thanks.